include/math/MultipliableScalarFunction.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef MultipliableScalarFunction_hpp
+#define MultipliableScalarFunction_hpp
+
+
+#include "math/ScalarFunctionProduct.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  An extension of a scalar function that implements its operator* most generally as returning a ScalarFunctionProduct, whose operator() is implemented as the product of the underlying operator()s
+ */
+template <typename _Valued, typename _Scalar, int _Cols>
+class MultipliableScalarFunction : public ScalarFunction<_Valued, _Scalar, _Cols> {
```

Comment:
Why does the `ScalarFunction` not implement the `*` operator? Could it cause issues if you want to implement smarter products?

Reply:
In general, what we call a scalar function in mathematics, would correspond to `MultipliableScalarFunction` because the evaluation of a product is the product of the evaluations. However, if we would implement `operator*` inside `ScalarFunction`, then a derived scalar function that has an easier way of implementing the product would have no means to do so because I think a derived class cannot overload a function with a different return type. Let me check on this.

Reply:
I've just checked, and I was confusing some things.

If `ScalarFunction` implements `ScalarFunctionProduct operator*(const ScalarFunction& rhs)`, then a derived `MyScalarFunction` can
- implements its own `MyScalarFunction operator*(const MyScalarFunction& rhs)` because the argument is different
- use `using ScalarFunction::operator*` to access the base `operator*()` from `ScalarFunction` to enable a multiplication with another scalar function.

I'll make sure to update the code.